client/data/disputes/actions.js

```diff
@@ -61,13 +108,265 @@ export function* acceptDispute( id ) {
 					dispute.order.number
 			  )
 			: __( 'You have accepted the dispute.', 'woocommerce-payments' );
-		yield dispatch( 'core/notices', 'createSuccessNotice', message );
+		yield dispatch( 'core/notices' ).createSuccessNotice( message );
 	} catch ( e ) {
 		const message = __(
 			'There has been an error accepting the dispute. Please try again later.',
 			'woocommerce-payments'
 		);
 		wcpayTracks.recordEvent( 'wcpay_dispute_accept_failed' );
-		yield dispatch( 'core/notices', 'createErrorNotice', message );
+		yield dispatch( 'core/notices' ).createErrorNotice( message );
+	}
+}
+
+function* handleSaveSuccess( id, submit ) {
+	const message = submit
+		? __( 'Evidence submitted!', 'woocommerce-payments' )
+		: __( 'Evidence saved!', 'woocommerce-payments' );
+	const href = addQueryArgs( 'admin.php', {
+		page: 'wc-admin',
+		path: '/payments/disputes',
+	} );
+
+	wcpayTracks.recordEvent(
+		submit
+			? 'wcpay_dispute_submit_evidence_success'
+			: 'wcpay_dispute_save_evidence_success'
+	);
+	/*
+			We rely on WC-Admin Transient notices to display success message.
+			https://github.com/woocommerce/woocommerce-admin/tree/master/client/layout/transient-notices.
+		*/
+	yield dispatch( 'core/notices' ).createSuccessNotice( message, {
+		actions: [
+			{
+				label: submit
+					? __( 'View submitted evidence', 'woocommerce-payments' )
+					: __(
+							'Return to evidence submission',
+							'woocommerce-payments'
+					  ),
+				url: addQueryArgs( 'admin.php', {
+					page: 'wc-admin',
+					path: '/payments/disputes/challenge',
+					id: id,
+				} ),
+			},
+		],
+	} );
+
+	getHistory().push( href );
+}
+
+function* handleSaveError( err, submit ) {
+	wcpayTracks.recordEvent(
+		submit
+			? 'wcpay_dispute_submit_evidence_failed'
+			: 'wcpay_dispute_save_evidence_failed'
+	);
+
+	const message = submit
+		? __( 'Failed to submit evidence. (%s)', 'woocommerce-payments' )
+		: __( 'Failed to save evidence. (%s)', 'woocommerce-payments' );
+	yield dispatch( 'core/notices' ).createErrorNotice(
+		sprintf( message, err.message )
+	);
+}
```

Comment:
I feel like this _might_ be moving too much functionality to the actions. Should these be component level functions?

I.e. should we use the return value from `submitEvidence` and `saveEvidence` to determine what kind of notice to display in the component as opposed to doing this in the action?

Reply:
Personally I think I prefer it here in the actions – the redirection, tracking, and notice are all external to the component.